sdk/resourcemanager/Azure.ResourceManager/src/Resources/Generated/ManagementGroupPolicyDefinitionResource.cs

```diff
@@ -126,7 +127,7 @@ public virtual Response<ManagementGroupPolicyDefinitionResource> Get(Cancellatio
                 var response = _managementGroupPolicyDefinitionPolicyDefinitionsRestClient.GetAtManagementGroup(Id.Parent.Name, Id.Name, cancellationToken);
                 if (response.Value == null)
                     throw new RequestFailedException(response.GetRawResponse());
-                return Response.FromValue(new ManagementGroupPolicyDefinitionResource(Client, response.Value), response.GetRawResponse());
```

Comment:
Yes this is a bit risky because we cannot guarantee that the type passed from the factory would be compatible to the real type.
But the generator might do not know the case that we are expecting it to return a derived resource, it only knows we want to convert from a resource data (from the rest operation class method) to a base resource (from the signature of this method), therefore it chooses this.
Let me think about this.

Reply:
I fixed this in the latest implementation